internal/cmd/version.go

```diff
@@ -108,50 +108,75 @@ func versionString() string {
 	return v
 }
 
-type versionCmd struct {
-	gs     *state.GlobalState
-	isJSON bool
-}
+// versionDetailsWithExtensions returns the structured details about version including extensions
+// returns error if there are unhandled extension types
+func versionDetailsWithExtensions(exts []*ext.Extension) (map[string]any, error) {
+	details := versionDetails()
 
-func (c *versionCmd) run(cmd *cobra.Command, _ []string) error {
-	if !c.isJSON {
-		root := cmd.Root()
-		root.SetArgs([]string{"--version"})
-		_ = root.Execute()
-		return nil
+	if len(exts) == 0 {
+		return details, nil
 	}
 
-	details := versionDetails()
-	if exts := ext.GetAll(); len(exts) > 0 {
-		type extInfo struct {
-			Module  string   `json:"module"`
-			Version string   `json:"version"`
-			Imports []string `json:"imports"`
-		}
+	// extInfo represents the JSON structure for an extension in the version details
+	// modeled after k6 extension registry structure
+	type extInfo struct {
+		Module  string   `json:"module"`
+		Version string   `json:"version"`
+		Imports []string `json:"imports,omitempty"`
+		Outputs []string `json:"outputs,omitempty"`
+	}
 
-		ext := make(map[string]extInfo)
-		for _, e := range exts {
-			key := e.Path + "@" + e.Version
+	infoList := make([]*extInfo, 0, len(exts))
+	infoMap := make(map[string]*extInfo)
 
-			if v, ok := ext[key]; ok {
-				v.Imports = append(v.Imports, e.Name)
-				ext[key] = v
-				continue
-			}
+	for _, e := range exts {
+		key := e.Path + "@" + e.Version
 
-			ext[key] = extInfo{
+		info, found := infoMap[key]
+		if !found {
+			info = &extInfo{
 				Module:  e.Path,
 				Version: e.Version,
-				Imports: []string{e.Name},
 			}
+
+			infoMap[key] = info
+			infoList = append(infoList, info)
 		}
 
-		list := make([]extInfo, 0, len(ext))
-		for _, v := range ext {
-			list = append(list, v)
+		switch e.Type {
+		case ext.OutputExtension:
+			info.Outputs = append(info.Outputs, e.Name)
+		case ext.JSExtension:
+			info.Imports = append(info.Imports, e.Name)
+		case ext.SecretSourceExtension:
```

Comment:
> LGTM, arguably we can have the secret source extension also in the output, but I will not block the PR on this

@mstoykov This "case" ensures that no error occurs. Whether we want an extra JSON property in the case of a secret source is another story.